### PR TITLE
Refactor app

### PR DESCRIPTION
**Pull request by pbochis**
Add behaviors to remove duplicate code.

### CODE REVIEW

File: .jshintrc [hunk 1]
```diff
@@ -69,12 +69,13 @@
...
-		"util"    : true,
-		"Chart"   : true
+		"app"     	: false,
+		"alert"   	: false,
+		"console" 	: false,
+		"page"    	: false,
+		"Polymer" 	: false,
+		"util"    	: true,
+		"Chart"   	: true,
+	  	"Behaviors"	: true
```
Comment (lorenzleutgeb) [inline .jshintrc:79]: Mixed spaces and tabs.

File: app/elements/services/user-service.html [hunk 2]
```diff
@@ -1,4 +1,6 @@
 <link rel="import" href="/bower_components/iron-ajax/iron-ajax.html">
+<link rel="import" href="/elements/behaviors/service-behavior.html">
+
```
Comment (lorenzleutgeb) [inline app/elements/services/user-service.html]: Pointless newline.

File: app/elements/behaviors/wsreader-behavior.html [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+		},
+		_onWsOpen: function(){
+			console.log('WS OPENED');
+		},
+		_onWsError: function(err){
+			console.log(err);
+		},
+		_onWsMessage: function(data){
+			//default implementation
+			this.$.console.setOutput(JSON.parse(data.detail));
```
Comment (victorbalan) [inline app/elements/behaviors/wsreader-behavior.html]: this will fail

File: app/elements/submission/tasks/general-code-task.html [hunk 4]
```diff
@@ -152,10 +153,12 @@
 				this.getTemplate(this.languages[0].tag);
 			},
 			languageChanged: function(){
+				console.log('someone is aware');
 				var languageTag = this.$.languageSelector.getSelectedLanguageTag();
 				this.getTemplate(languageTag);
 			},
 			getTemplate: function(language){
+				console.log('un template ' + language);
```
Comment (lorenzleutgeb) [inline app/elements/submission/tasks/general-code-task.html]: d

File: app/elements/submission/tasks/general-code-task.html [hunk 5]
```diff
@@ -152,10 +153,12 @@
 				this.getTemplate(this.languages[0].tag);
 			},
 			languageChanged: function(){
+				console.log('someone is aware');
```
Comment (lorenzleutgeb) [inline app/elements/submission/tasks/general-code-task.html]: d

File: app/elements/submission/tasks/general-code-task.html [hunk 6]
```diff
@@ -123,6 +123,7 @@
 			],
 			afterPropertiesSet: function(){
 				this.languages = [];
+				console.log('dafuq');
```
Comment (lorenzleutgeb) [inline app/elements/submission/tasks/general-code-task.html]: d

**Pull request merged**